core/modules/sql/SqlConnection.cc

```diff
@@ -126,14 +127,29 @@ SqlConnection::~SqlConnection() {
 
 bool
 SqlConnection::connectToDb(SqlErrorObject& errObj) {
-    assert(_connection.get());
-    if (_connection->connected()) return true;
+    assert(_connection != nullptr);
+    static int lastChecked = 0;
+    time_t currentTime = time(0);
+
+    if (_connection->connected()) {
+        int rc = mysql_ping(_connection->getMySql());
+        if (rc == 0) {
+            if (currentTime < lastChecked+60) return true;
```

Comment:
Sorry, ignore this comment, I'm confused by trying to guess intention here. Code as it is written does not do much useful, as John says currentTime and lastChecked can be removed. My first guess was that you want to limit ping frequency, but ping is done every time anyways.